ARE/services/WebService/src/main/java/eu/asterics/mw/webservice/serverUtils/ServerRepository.java

```diff
@@ -35,17 +39,98 @@
 public class ServerRepository {
+    //Singleton instance
+    private static ServerRepository instance;
+    
+	
 
     // Rest Server configuration
+    public static final String ARE_WEBSERVICE_PORT_REST_KEY = "ARE.webservice.port.REST";
     public static final String PATH_REST = "/rest";
-    public static final int PORT_REST = 8081;
-    public static final URI BASE_URI_REST = URI.create("http://0.0.0.0:" + PORT_REST + PATH_REST);
-
-    // Web Service Server configuration
-    public static final String PATH_WS = "/ws";
-    public static final String PATH_WS_ASTERICS_DATA = "/astericsData";
-    public static final int PORT_WS = 8082;
-    public static final URI BASE_URI_WS = URI.create("http://localhost:" + PORT_WS + PATH_WS);
+    public static final int DEFAULT_PORT_REST = 8081;    
+    
+    // Web Socket Server configuration
+    public static final String PATH_WEBSOCKET = "/ws";
+    public static final String PATH_WEBSOCKET_ASTERICS_DATA = "/astericsData";
+    public static final String ARE_WEBSERVICE_PORT_WEBSOCKET_KEY = "ARE.webservice.port.websocket";    
+    public static final int DEFAULT_PORT_WEBSOCKET = 8082;
+    
+    //member variables holding property values
+    private int portREST=DEFAULT_PORT_REST;
+    private int portWebsocket=DEFAULT_PORT_WEBSOCKET;
+    
+    /**
+     * Private ctor used for initializing the class.
+     */
+    private ServerRepository() {
+    	//init ports and paths with property values
+    	try {
+    		portREST=Integer.parseInt(AREProperties.instance.getProperty(ARE_WEBSERVICE_PORT_REST_KEY, String.valueOf(DEFAULT_PORT_REST)));
+    		//We have to store back the actual property value so that it will be written to the areProperties config file if it did not exist before.
+    		AREProperties.instance.setProperty(ARE_WEBSERVICE_PORT_REST_KEY, String.valueOf(portREST));
+    	}catch(NumberFormatException e) {
+    		AstericsErrorHandling.instance.getLogger().logp(Level.WARNING, this.getClass().getName(), "ServerRepository()", "Configured port for REST service invalid: "+ e.getMessage(),e);
+    	}
+    	//init ports and paths with property values
+    	try {
+    		portWebsocket=Integer.parseInt(AREProperties.instance.getProperty(ARE_WEBSERVICE_PORT_WEBSOCKET_KEY, String.valueOf(DEFAULT_PORT_WEBSOCKET)));
+    		//We have to store back the actual property value so that it will be written to the areProperties config file if it did not exist before.
+    		AREProperties.instance.setProperty(ARE_WEBSERVICE_PORT_WEBSOCKET_KEY, String.valueOf(portWebsocket));
+    	}catch(NumberFormatException e) {
+    		AstericsErrorHandling.instance.getLogger().logp(Level.WARNING, this.getClass().getName(), "ServerRepository()", "Configured port for Websocket service invalid: "+e.getMessage(),e);
+    	}
+    }
+    
+    /**
+     * Returns the baseURI for the REST API.
+	 * @return the baseUriRest
+	 */
+	public URI getBaseUriREST() {
+		return URI.create("http://0.0.0.0:" + getPortREST() + PATH_REST);
```

Comment:
is there a reason that we have 0.0.0.0 here and not localhost?

Reply:
The 0.0.0.0 means that it shall be registered for all network interfaces, so that the webserver is reachable from localhost and from another machine as well.

Reply:
ok, so if it would be "localhost" the webserver would not be reachable from other clients in the network? i'm wondering because the websocket is also initialized with "localhost" and websocket-communication is nevertheless possible from remote devices.